pyproject.toml

```diff
@@ -95,6 +95,7 @@ open_files_ignore = ['test.fits', 'asdf.fits']
 filterwarnings = [
     'error',
     'ignore::asdf.exceptions.AsdfDeprecationWarning:asdf.asdftypes',
+    'ignore:.*subclasses the deprecated CustomType:asdf.exceptions.AsdfDeprecationWarning:asdf.types',
```

Comment:
How difficult would it be for us explicitly catch these warnings in our tests? This will mark the tests that need to be adjusted/removed when `CustomType` is fully removed.

Reply:
The issue I ran into trying per-test ignores is that warnings turn into errors during test collection and no tests get run. Is there a way to work around this?
One thought I had was to remove the `error` filter from pyproject.toml and add a pytest fixture that turned warnings into errors but this seemed more confusing than just adding the warning here.
Other PRs adding deprecations have similar issues so finding an elegant fix for this would be good.

Reply:
Adding this deprecation warning will require way more thought than what is being done here. The deprecation warning will be raised by ASDF every single time it tries to load entry points, which is unacceptable to me.

Reply:
Note that for most of the things raising the deprecation error on startup this can be solved by adding a fixture which loads and catches the warning.

Reply:
> Adding this deprecation warning will require way more thought than what is being done here. The deprecation warning will be raised by ASDF every single time it tries to load entry points, which is unacceptable to me.

Perhaps I am not understanding what you are describing. I am only seeing the warning the first time the entry points are loaded:
```python
import warnings

import asdf.entry_points

with warnings.catch_warnings():
    warnings.simplefilter("ignore")
    _ = asdf.entry_points.get_extensions()

_ = asdf.entry_points.get_extensions()
```
The above run with `python -X dev` shows no warnings for the second call to get_extensions. Would you provide an example of what behavior your are objecting to?

The deprecation warning is raised once per subclass of CustomType to inform the developer which classes require updating.

Reply:
I amend my above statement. The issue is when astropy's extension gets loaded, it issues a bunch of warnings these will all need to be ignored.